src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -535,21 +553,31 @@ def lsm_service_catalog_get_entity_version(
         version: int,
     ) -> inmanta.protocol.common.Result:
         """
-        This is not supported yet by LsmProject mock.  Until then, behaves as if the
-        api didn't know this endpoint (lsm module will fallback to the legacy one).
+        This is a mock for the lsm api, this method is called when `include_purged_embedded_entities` is set to true
+        when unrolling a service entity binding.
         """
-        # https://github.com/inmanta/pytest-inmanta-lsm/issues/467
+        assert (
+            self.service_entities is not None
+        ), "The service catalog has not been initialized, please call self.export_service_entities"
+        assert str(tid) == self.environment, f"{tid} != {self.environment}"
+
+        if (service_entity, version) not in self.service_entities:
+            return inmanta.protocol.common.Result(code=404)
+
         return inmanta.protocol.common.Result(
-            code=500,
+            code=200,
             result={
-                "message": "LsmProject doesn't support multi-version lsm yet",
+                "data": json.loads(
+                    json.dumps(
+                        self.service_entities[(service_entity, version)],
+                        default=inmanta.util.api_boundary_json_encoder,
+                    ),
+                ),
             },
         )
 
     def lsm_service_catalog_get_entity(
-        self,
-        tid: uuid.UUID,
-        service_entity: str,
+        self, tid: uuid.UUID, service_entity: str, version: typing.Optional[int] = None
```

Comment:
```suggestion
        self, tid: uuid.UUID, service_entity: str, version: int | None = None
```
Why use the old notation? This code base is python3.11+ (iso6 is dead)

(also applies in other places in this file)

Reply:
Force of habit. Updated it!